tools/release_builder.py

```diff
@@ -74,33 +74,81 @@ def _run_cmd(self, cmd, exit_on_fail=True, dry_run_return = 0):
+            pkgdir_path = os.path.join(
+                scratchdir,
+                'stub-universe-{}'.format(self._pkg_name),
+                'repo',
+                'packages',
+                self._pkg_name[0].upper(),
+                self._pkg_name,
+                '0')
+            if not os.path.isdir(pkgdir_path):
+                raise Exception('Didn\'t find expected path {} after unzipping {}'.format(
+                    pkgdir_path, self._stub_universe_url))
+            return pkgdir_path
+        elif stub_universe_extension == '.json':
+            # stub universe json file (universe 3.x+ only)
+            stub_universe_json = json.loads(stub_universe_file.read().decode('utf-8'), object_pairs_hook=OrderedDict)
+
+            # put package files into a subdir of scratchdir: avoids conflicts with reuse of scratchdir elsewhere
+            pkgdir = os.path.join(scratchdir, 'stub-universe-{}'.format(self._pkg_name))
+            os.makedirs(pkgdir)
+
+            if 'packages' not in stub_universe_json:
+                raise Exception('Expected "packages" key in root of stub universe JSON: {}'.format(
+                    self._stub_universe_url))
+            if len(stub_universe_json['packages']) != 1:
+                raise Exception('Expected single "packages" entry in stub universe JSON: {}'.format(
+                    self._stub_universe_url))
+
+            package_json = stub_universe_json['packages'][0]
+
+            def extract_json_file(package_dict, name):
+                file_dict = package_dict.get(name, None)
+                if file_dict is not None:
+                    del package_dict[name]
+                    # ensure that the file has a trailing newline (json.dump() doesn't!)
+                    fileref = open(os.path.join(pkgdir, name + '.json'), 'w')
+                    content = json.dumps(file_dict, indent=2)
+                    fileref.write(content)
+                    if not content.endswith('\n'):
+                        fileref.write('\n')
+                    fileref.flush()
+                    fileref.close()
+            extract_json_file(package_json, 'command')
+            extract_json_file(package_json, 'config')
+            extract_json_file(package_json, 'resource')
+
+            marathon_json = package_json.get('marathon', {}).get('v2AppMustacheTemplate', None)
+            if marathon_json is not None:
+                del package_json['marathon']
+                marathon_file = open(os.path.join(pkgdir, 'marathon.json.mustache'), 'w')
+                marathon_file.write(base64.standard_b64decode(marathon_json).decode())
+                marathon_file.flush()
+                marathon_file.close()
+
+            if 'releaseVersion' in package_json:
+                del package_json['releaseVersion']
+
+            json.dump(package_json, open(os.path.join(pkgdir, 'package.json'), 'w'), indent=2)
```

Comment:
And now the destructive handling becomes clear.  Maybe worth a brief note at the top?